mainwindow.cpp

```diff
@@ -123,17 +124,19 @@ MainWindow::MainWindow(QWidget *parent, const QString &session)
 
     // define an action to delete the selected row from the list
     connect(removeSelected, &QAction::triggered, m_command_history, [=]() {
-        int currentRow = m_command_history->currentRow();
-        if (-1 != currentRow) {
-            delete m_command_history->takeItem(currentRow);
+        if (true == m_command_history->selectionModel()->hasSelection()) {
+            QList<QModelIndex> selectedItems = m_command_history->selectionModel()->selectedIndexes();
+            for (auto item = selectedItems.crbegin(); item != selectedItems.crend(); ++item) {
```

Comment:
This makes the code dependant on Qt 5.6 and above. (http://doc.qt.io/qt-5/qlist.html#crbegin)
I don't think this is appropriate as of yet.